test/models/rss_log_test.rb

```diff
@@ -2,5 +2,44 @@
 
 require("test_helper")
 
+# tests of MO's "rss_logs"
 class RssLogTest < UnitTestCase
+  def test_url_for_normalized_controllers
+    normalized_rss_log_types.each do |type|
+      rss_log = rss_log_for_type(type)
+      id = rss_log.target_id
+      assert_match(type_normalized_show_path(type, id), rss_log.url,
+                   "rss_log.url incorrect for #{model(type)}")
+    end
+  end
+
+  # ---------- helpers ---------------------------------------------------------
+
+  def normalized_rss_log_types
+    RssLog.all_types.each_with_object([]) do |type, ary|
+      ary << type if model(type).controller_normalized?(model(type).name)
```

Comment:
Why does model(type).name have to be passed as a parameter?

Reply:
I know that code looks (and probably is) ridiculous. But`AbstractModel#controller_normalized?` takes an argument.
```ruby
  def self.controller_normalized?(name)
    class_defined?("#{name.pluralize}Controller")
  end
```
I no longer remember why I wrote `controller_normalized?` that way.